Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift

```diff
@@ -205,14 +205,22 @@ extension BrowserViewController: WKNavigationDelegate {
             return
         }
 
-        if let tab = tabManager.selectedTab, isStoreURL(url) {
+        if isStoreURL(url) {
             decisionHandler(.cancel)
 
-            let alreadyShowingSnackbarOnThisTab = tab.bars.count > 0
-            if !alreadyShowingSnackbarOnThisTab {
-                TimerSnackBar.showAppStoreConfirmationBar(forTab: tab, appStoreURL: url)
+            // Make sure to wait longer than delaySelectingNewPopupTab to ensure selectedTab is correct
+            DispatchQueue.main.asyncAfter(deadline: .now() + tabManager.delaySelectingNewPopupTab + 0.1) {
+                guard let tab = self.tabManager.selectedTab else { return }
+                let alreadyShowingSnackbarOnThisTab = tab.bars.count > 0
```

Comment:
i found the logic here a bit convoluted. I think doing `if tab.bars.isEmpty` is pretty clear